internal/exec/exec.go

```diff
@@ -62,6 +62,71 @@ type fieldToExec struct {
 	out      *bytes.Buffer
 }
 
+// TODO: Don't want to pass request here, but need the directive visitors exposed on the field
+func (f *fieldToExec) resolve(ctx context.Context, r *Request) (output interface{}, err error) {
```

Comment:
I agree with you, don't pass the request here! The visitors should not even be a field of the `Request` struct. See how the field args are in an `ArgsPacker` of type `reflect.Value`. Calling the `Interface()` method on that returns the actual value. I believe we need "packed directive visitors as well"

Reply:
The code that adds the `ArgsPacker` should create an instance of the directive visitor as well (imagine a field named `DirectivesPacker`). In your code you are reusing the same directive visitor instance for all fields that have this directive. Instead, you need to create a unique instance for every occurrence of the directive for every field. Sometimes, even more than once per field in the case of repeatable directives.

Reply:
Now implemented